src/Hooks/useReportNFT/useReportNFT.ts

```diff
@@ -1,50 +1,33 @@
 import { useNavigation } from "@react-navigation/native"
-import { NativeStackNavigationProp } from "@react-navigation/native-stack"
 import { useCallback } from "react"
-import { abi, Transaction } from "thor-devkit"
-import { abis, AnalyticsEvent } from "~Constants"
-import { NFT_BLACKLIST_CONTRACT, NFT_BLACKLIST_CONTRACT_TESTNET } from "~Constants/Constants/NFT"
+import { AnalyticsEvent } from "~Constants"
 import { useAnalyticTracking } from "~Hooks/useAnalyticTracking"
-import { useBlockchainNetwork } from "~Hooks/useBlockchainNetwork"
-import { RootStackParamListNFT, Routes } from "~Navigation"
+import { useNFTReportTransaction } from "~Hooks/useNFTReportTransaction"
+import { Routes } from "~Navigation"
 
 export const useReportNFT = () => {
-    const nav = useNavigation<NativeStackNavigationProp<RootStackParamListNFT>>()
+    const navigation = useNavigation()
     const track = useAnalyticTracking()
-    const { isMainnet } = useBlockchainNetwork()
-
-    const blacklistContract = isMainnet ? NFT_BLACKLIST_CONTRACT : NFT_BLACKLIST_CONTRACT_TESTNET
-
-    const buildReportTxClause = useCallback(
-        (nftAddress: string): Transaction.Clause[] => {
-            const reportAbi = abis.NFTBlacklist.NFTBlacklistAbis.report
-            if (!reportAbi) throw new Error("Function abi not found for reporting NFT collection")
-            const reportData = new abi.Function(reportAbi).encode(nftAddress)
-            return [
-                {
-                    to: blacklistContract,
-                    data: reportData,
-                    value: "0x0",
-                },
-            ]
-        },
-        [blacklistContract],
-    )
+    const { buildReportClause } = useNFTReportTransaction()
 
     const reportNFTCollection = useCallback(
         (nftAddress: string) => {
-            const clause = buildReportTxClause(nftAddress)
+            const clause = buildReportClause(nftAddress)
 
             track(AnalyticsEvent.NFT_COLLECTION_REPORTED, {
                 nftAddress: nftAddress,
             })
 
-            nav.replace(Routes.REPORT_NFT_TRANSACTION_SCREEN, {
-                nftAddress,
-                transactionClauses: clause,
+            // @ts-ignore - Cross-stack navigation typing
+            navigation.navigate(Routes.NFTS, {
+                screen: Routes.REPORT_NFT_TRANSACTION_SCREEN,
+                params: {
+                    nftAddress,
+                    transactionClauses: clause,
+                },
             })
```

Comment:
I see that this screen has been added to the home stack, so it doesn't need a cross navigation.
Also the NFT Stack will be deprecated with the collectibles